app/validators/age_validator.rb

```diff
@@ -0,0 +1,39 @@
+class AgeValidator < ActiveModel::EachValidator
+  MAX_AGE = 120
+
+  def validate_each(record, attribute, value)
+    if value.present?
+      check_is_date(record, attribute, value) &&
+        check_range(record, attribute, value)
+    end
+  end
+
+private
+
+  def check_is_date(record, attribute, value)
+    if value.is_a?(Date)
+      true
+    else
+      record.errors.add(
+        attribute,
+        I18n.t('age_validator.errors.invalid_date')
```

Comment:
Yes that issues a translation missing.
@alan   Technically you don't need the `I18n.t` but you have to insert the key under the relevant ActiveModel model (i.e en.errors.active_model.model.model_name.atttributes.attiribute_name...) . Since this an isolated validator and in theory can be re-used everywhere this is perfectly fine since it decouples the validator from the models it is included in.